plugins/flespiConnector.ts

```diff
@@ -116,31 +114,34 @@ function createClient (): MqttClient {
 function setupClient (client: MqttClient, channels: Channel[]): MqttClient {
     // When the client is connected, we subscribe to the telemetry topic
     client.on("connect", () => {
-        console.log(`Connected ${client.options.clientId}`);
+        console.log("Connected...");
         channels.forEach((channel: Channel) => {
             client.subscribe("flespi/message/gw/channels/" + channel.id + "/+", { qos: 1 }, (err: Error) => {
                 if (err) {
-                    console.log("testttt");
                     client.end(true); // force disconnect
                 }
-                console.log(`Subscribed to channel ${channel.id}`);
             });
         });
     });
 
     // emits new coordinates whenever the subscription receives new data
     client.on("message", (topic: string, msg: Buffer) => {
-        console.log(topic, msg.toString("utf8"));
-        /*
-         * const splitTopic: string[] = topic.split("/");
-         * const informationName: string = splitTopic[splitTopic.length - 1];
-         * const deviceID: number = Number(splitTopic[splitTopic.length - 3]); // Get the device ID
-         *
-         * if (informationName === "position") {
-         *  // emitNewCoordinates(convertToObject(msg.toString("utf8"))); // Emit the new position
-         *  console.log("emit new coordinates");
-         * }
-         */
+        console.log("message");
+        const splitTopic: string[] = topic.split("/");
+        const locationId: string = splitTopic[splitTopic.length - 1];
+        const data = JSON.parse(msg.toString("utf-8"));
+        emitNewCoordinates(locationId, {
+            latitude: data["position.latitude"],
+            longitude: data["position.longitude"]
+        });
+    });
+
+    client.on("error", () => {
+        client.end(true);
+    });
+
+    client.on("close", () => {
+        client.end(true);
```

Comment:
I think this is not needed!
The `close` event happens AFTER the session has ended.

```suggestion
        console.log("Disconnected");
```